### PR TITLE
Update "Build blockchain" overview page

### LINKED ISSUE #1518: Update `Building a blockchain application`
### Description

* Update `Building a blockchain application` to use v6 interfaces
* Update terminologies to Lisk SDK v6

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1518

### CODE REVIEW

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 1]
```diff
@@ -1,52 +1,48 @@
...
-These guides explain how to include and use different plugins in the application.
 
-* xref:{url_guide_faucet}[]
-* xref:{url_guide_dashboard}[]
+The following image provides an overview of the "Hello" application:
 
-=== Other
-* xref:{url_guide_launch}[]
-* xref:{url_guide_test}[]
+image::build-blockchain/hello-client.png["Hello client overview", 400]
```
Comment (chris529) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc:48]: imho I would reword the one line in the Stores box to read as follows:

"Is of a maximum length of 300 characters".

**Review by chris529 — APPROVED** [review]
LGTM! 👍

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 2]
```diff
@@ -1,52 +1,48 @@
-= Building blockchain applications with the Lisk SDK
+= Building a sidechain with the Lisk SDK
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc:1]: I think we should say 'sidechain client' instead of a sidechain.
```suggestion
= Building a sidechain client
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 3]
```diff
@@ -1,52 +1,48 @@
...
-:url_guide_quickstart: lisk-sdk::quickstart.adoc
-:url_guide_faucet: build-blockchain/enable-faucet.adoc
-:url_guide_dashboard: build-blockchain/using-dashboard.adoc
-:url_guide_launch: build-blockchain/launch-test-network.adoc
-:url_guide_test: build-blockchain/test-app.adoc
-
-The "Build Blockchain" category contains all the guides related to the development of a blockchain application.
 
-== Quickstart
+The "Build Blockchain" category contains all the guides related to the development of a sidechain client.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
The "Build Blockchain" category contains all the guides related to the development of a sidechain client with the Lisk SDK.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 4]
```diff
@@ -1,52 +1,48 @@
...
 
-The xref:{url_guide_quickstart}[] guide describes in a few simple steps how to set up a blockchain application and how to start it successfully for the first time.
+== The Hello World sidechain
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
== The "Hello World" client
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 5]
```diff
@@ -1,52 +1,48 @@
...
 
-== Guides for building blockchain applications
+By the example of the "Hello World" client, it is explained how to build a very simple sidechain client step by step in the following chapters.
 
-=== The Hello World application
+NOTE: Although the example is kept very simple by intention, the Hello World client implements and uses all the most important components of a module, and is therefore a good example to understand the basics of module development.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
NOTE: Although the example is kept very simple by intention, the Hello World client implements and uses all the most important components of a client such as modules, plugins, configurations, genesis blocks, etc., and is therefore a good example to understand the basics of a client's development.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 6]
```diff
@@ -1,52 +1,48 @@
...
 
-How to create a very simple blockchain application by using the example of the "Hello World" blockchain application, which allows users to save hello messages in their accounts.
+The "Hello World" sidechain allows each account to save a greeting message on-chain.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
The "Hello World" sidechain client allows each account to save a greeting message on-chain.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 7]
```diff
@@ -1,52 +1,48 @@
...
 
 The following guides each focus on developing a specific part of the Hello application.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
The following guides each focus on developing a specific part of the Hello client.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 8]
```diff
@@ -1,52 +1,48 @@
...
-image:guides/hello-overview.png[align="center"Hello application overview, 600]
+. xref:{url_guide_app}[]: How to bootstrap a default sidechain client.
+. xref:{url_guide_config}[]: How to create a custom configuration for a sidechain client.
+. xref:{url_guide_module}[]: How to create a new module and how to add it to the client.
+In the example, the `HelloModule` will contain all on-chain logic related to the Hello message.
+.. xref:{url_guide_command}[]: How to create a new command and how to add it to a module.
+In the example, the "Create Hello" command saves Hello messages of accounts, if they send a corresponding transaction.
+. xref:{url_guide_genesisblock}[]: How to generate a new genesis block for a sidechain.
+. xref:{url_guide_plugin}[]: How to create a new plugin and how to add it to the sidechain client.
+In the example, the plugin will return the latest Hello message which was sent to the network, and its' sender address.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
In the example, the plugin will return the addresses of all the senders who sent a Hello Message to the network.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 1 above)
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc:48]: - Please update the description of the plugin as per line 37.
- Also, the plugin will have DB, Endpoint, Schemas, and Configurations, and since this is the overview page of the sidechain client, I think these elements should go into the diagram as well.

File: docs/modules/ROOT/pages/build-blockchain/index.adoc [hunk 9]
```diff
@@ -1,52 +1,48 @@
...
+== The Hello World sidechain
 
-== Guides for building blockchain applications
+By the example of the "Hello World" client, it is explained how to build a very simple sidechain client step by step in the following chapters.
 
-=== The Hello World application
+NOTE: Although the example is kept very simple by intention, the Hello World client implements and uses all the most important components of a module, and is therefore a good example to understand the basics of module development.
 
-How to create a very simple blockchain application by using the example of the "Hello World" blockchain application, which allows users to save hello messages in their accounts.
+The "Hello World" sidechain allows each account to save a greeting message on-chain.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
It also stores the addresses of senders of the greeting message, in an off-chain database.
```

**Review by TalhaMaliktz — CHANGES_REQUESTED** [review]
Looks good! A few suggestions. . . .  :)

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 2 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc:1]: I disagree, I think "sidechain" is more fitting in the context, because this cateogry will also include guides how to register the app as a sidechain. So imho it should be either "sidechain" or "sidechain application" in the title.

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 5 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
NOTE: Although the example is kept very simple by intention, the Hello World client implements all the most important components, and is therefore a good example to understand the basics of the development of a sidechain client.
```

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 8 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc]: Will it only return the adresses or the Hello messages as well?

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 1 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc:48]: @TalhaMaliktz could you update the plugin part of the diagram on Lucidchart? I can then update the diagram in the PR, or it can be updated again in the PR that updates the "How to create a plugin" guide

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 2 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc:1]: Yeah in that case, sidechain application sounds good to me.

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 8 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc]: We can return both as long they are returned in the event, haven't decoded the event's data yet.

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 1 above)
Comment (TalhaMaliktz) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc:48]: Sure, will update it during my "How to create a plugin" PR

File: docs/modules/ROOT/pages/build-blockchain/index.adoc (hunk 8 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/index.adoc]:
```suggestion
In the example, the plugin will save all sent Hello messages and their corresponding sender address in an off-chain store, and provide endpoints to retrieve the data from the store.
```

**Review by TalhaMaliktz — APPROVED** [review]
LGTM 👍🏼
